### ISSUE TITLE
feat: add service worker module

### ISSUE THREAD

**Issue by pemrouz** [rijs/fullstack#27]

**Comment by 3liv** [comment]
I've been looking at this. Only chrome supports background sync and would need to update offline module to use indexedDB as localstorage not available from service workers

**Comment by pemrouz** [comment]
Agreed 👍 , this should be a nice progressive enhancement that will automatically boost apps when used in a browser that supports ServiceWorker (can use the same cache header info). If there is no ServiceWorker, the offline module should still use the localStorage trick as that still goes a long way in boosting perceived load times in current browsers.
